src/charm.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+import logging
+from urllib.request import urlopen
+
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import BlockedStatus, MaintenanceStatus
+from charms.operator_libs_linux.v0 import apt
+
+MONGO_GPG_URL = "https://www.mongodb.org/static/pgp/server-5.0.asc"
+MONGO_REPO = "https://repo.mongodb.org/apt/ubuntu"
+
+logger = logging.getLogger(__name__)
+
+
+class MongodbOperatorCharm(CharmBase):
+    """Charm the service."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.install, self._on_install)
+
+    def _on_install(self, _):
+        """Install prerequisites for the application"""
+        self.unit.status = MaintenanceStatus("installing mongodb")
+        self._add_mongodb_org_repository()
+        self._install_apt_packages(["mongodb-org"])
+
+    def _add_mongodb_org_repository(self):
+        repositories = apt.RepositoryMapping()
+
+        # Get GPG key
+        key = urlopen("https://www.mongodb.org/static/pgp/server-5.0.asc").read().decode()
```

Comment:
We may want some exception handling here

Reply:
I see you have `MONGO_GPG_URL`, maybe you can use it here or not have it declared at all, the linter will complain about that